CodePush.js

```diff
@@ -170,8 +183,33 @@ function setUpTestDependencies(testSdk, providedTestConfig, testNativeBridge) {
   if (testNativeBridge) NativeCodePush = testNativeBridge;
 }
 
+// This function allows only one syncInternal operation to proceed at any given time.
+// Parallel calls to sync() while one is ongoing yields CodePush.SyncStatus.SYNC_IN_PROGRESS.
+const sync = (() => {
+  let syncInProgress = false;
+  const setSyncCompleted = () => { syncInProgress = false; };
+  
+  return (options = {}, syncStatusChangeCallback, downloadProgressCallback) => {
+    if (syncInProgress) {
+      typeof syncStatusChangeCallback == "function"
+        ? syncStatusChangeCallback(CodePush.SyncStatus.SYNC_IN_PROGRESS)
+        : log("Sync already in progress.");
```

Comment:
I'd prefer always logging this, instead of only when there isn't a sync callback. That way, basic diagnostics could still be done via the console as opposed to having to set break points or expect the dev to add their own logging.